daemon/cmd/datapath.go

```diff
@@ -198,91 +198,73 @@ func (d *Daemon) syncEndpointsAndHostIPs() error {
 		return nil
 	}
 
-	specialIdentities := []identity.IPIdentityPair{}
+	existingEndpoints, err := lxcmap.DumpToMap()
+	if err != nil {
+		return err
+	}
 
 	if option.Config.EnableIPv4 {
-		addrs, err := d.datapath.LocalNodeAddressing().IPv4().LocalAddresses()
-		if err != nil {
-			log.WithError(err).Warning("Unable to list local IPv4 addresses")
-		}
-
-		for _, ip := range addrs {
-			if option.Config.IsExcludedLocalAddress(ip) {
-				continue
+		naf := d.datapath.LocalNodeAddressing().IPv4()
+		overrides := naf.SubnetOverride()
+
+		var err error
+		if len(overrides) > 0 {
+			for _, ipNet := range overrides {
+				err = upsertIPIdentity(identity.IPIdentityPair{
+					IP:   ipNet.IP,
+					Mask: ipNet.Mask,
+					ID:   identity.ReservedIdentityHost,
+				}, &existingEndpoints)
+				if err != nil {
+					break
+				}
 			}
-
-			if len(ip) > 0 {
-				specialIdentities = append(specialIdentities,
-					identity.IPIdentityPair{
-						IP: ip,
-						ID: identity.ReservedIdentityHost,
-					})
+		} else {
+			err = ipAddrMapFn(naf, &existingEndpoints)
+			if err != nil {
+				err = upsertIPIdentity(identity.IPIdentityPair{
+					IP:   net.IPv4zero,
+					Mask: net.CIDRMask(0, net.IPv4len*8),
+					ID:   identity.ReservedIdentityWorld,
+				}, &existingEndpoints)
 			}
```

Comment:
This block here overwrites the initial error on `ipAddrMapFn` with whatever happens with `upsertIPIdentity`. Is that the behavior we intend? Are there cases where we wouldn't want this behavior?